pkg/webui/oauth/views/update-password/index.js

```diff
@@ -0,0 +1,221 @@
+    user: PropTypes.object,
+    handleCancelUpdate: PropTypes.func.isRequired,
+    push: PropTypes.func.isRequired,
+  }
+
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      error: '',
+      info: '',
+      revoke_all_access: false,
+    }
+  }
+
+  handleRevokeAllAccess (evt) {
+    this.setState({ revoke_all_access: evt.target.checked })
+  }
+
+  async handleSubmit (values, { setSubmitting }) {
+    const { user, push } = this.props
+    const userParams = queryString.parse(this.props.location.search)
+    const oldPassword = values.old_password ? values.old_password : userParams.current
+    const userId = Boolean(user) ? user.ids.user_id : userParams.user
+    try {
+      await api.users.updatePassword( userId, {
+        user_ids: { user_id: userId },
+        'new': values.password,
+        old: oldPassword,
+        revoke_all_access: values.revoke_all_access,
+      })
+      if (Boolean(user)) {
+        this.setState({
+          error: '',
+          info: m.passwordChanged,
+          changed: true,
+        })
+      } else {
+        push('/login', {
+          info: m.passwordChanged,
+        })
+      }
+    } catch (error) {
+      this.setState({
+        error: error.response.data,
+        info: '',
+      })
+    } finally {
+      setSubmitting(false)
+    }
+  }
+
+  handleCancel () {
+    this.props.handleCancelUpdate()
+  }
+
+  render () {
+    const {
+      user,
+      fetching,
+    } = this.props
+
+    const {
+      error,
+      info,
+      revoke_all_access,
+    } = this.state
+
+    let oldPasswordField
+
+    if (fetching) {
+      return (
+        <Spinner center>
+          <Message content={sharedMessages.loading} />
+        </Spinner>
+      )
+    }
+
+    if (Boolean(user)) {
+      oldPasswordField = (
+        <Form.Field
+          component={Input}
+          required
+          title={m.oldPassword}
+          name="old_password"
+          type="password"
+          autoComplete="old-password"
+          autoFocus
+        />
+      )
+    }
+
+    return (
+      <Container className={style.fullHeight}>
+        <Row justify="center" align="center" className={style.fullHeight}>
+          <Col sm={12} md={8} lg={5}>
+            <IntlHelmet title={sharedMessages.changePassword} />
+            <Message content={sharedMessages.changePassword} component="h1" className={style.title} />
+            <Form
+              onSubmit={this.handleSubmit}
+              initialValues={initialValues}
+              error={error}
+              info={info}
+              validationSchema={validationSchema}
+              horizontal={false}
+            >
+              {oldPasswordField}
+              <Form.Field
+                component={Input}
+                required
+                title={m.newPassword}
+                name="password"
+                type="password"
+                autoComplete="new-password"
+                autoFocus
```

Comment:
Autofocus should not be set when the olf password field is rendered. In that case, the old password field should have the autofocus.